accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,126 @@
+
+To facilitate this, we will package analyzers inside [reference packs](https://github.com/dotnet/designs/blob/main/accepted/2019/targeting-packs-and-runtime-packs.md).  We will use the standard [nuget conventions for describing analyzers](https://github.com/dotnet/designs/blob/main/accepted/2019/targeting-packs-and-runtime-packs.md).  The .NET SDK will be responsible for locating the appropriate analyzers for the TargetFramework and creating `@(Analyzer)` items as part of the `ResolveTargetingPackAssets` task.  These items will be conflict-resolved with other Analyzer sources based as described in the following section.
+
+### Source generator conflict resolution
+
+Source generators and analyzers are passed to the compiler as @(Analyzer) items.  These items may come from NuGet packages, be built into the SDK targets, be directly defined by the user via ProjectReference or raw Analyzer items, or be provided by the new framework mechanism described in this document.  All should be considered for conflict resolution based on the same constraint.
+
+The .NET SDK already does conflict resolution for other asset types: references and copy-local runtime assets.  It should do the same for Analyzers.  Analyzers need not have "platform manifest" inputs for conflict resolution since all analyzers will be available during build, unlike runtime files which may not be available.  Comparison for the sake of conflict resolution can consider the same assembly and file version rules as reference files.  When finding two or more source generators with the same name
+
+Source generator conflict resolution should run before compile, but after all NuGet package assets have been evaluated.  Specific sequencing, public/private target naming, extensibility will not be specified here but should follow best practice as determined by the SDK team. 
+
+### Source generator versioning
+
+Source generators should be strongly named and update assembly version with every public stable release (GA and servicing).  This ensures that the publicly shipped source generators can be loaded side-by-side in the same process and garuntees 
+
+### Source generator compatibility concerns
+
+Source generators should only use public API in framework components they extend.  This public API already needs to meet the strict binary compatibility guarantees as framework API.  As such libraries containing source-generated code will be binary-compatible with future releases, just as if the user wrote this code directly.
```

Comment:
> Source generators should only use public API in framework components they extend

But should be able to build source used by the framework component into the source generator as well.  For example, regex's parser is non-public API and we can build the source for it into both Regex and the regex source generator.  This doesn't create a dependency between them any more than implementing a custom parser for the source generator does.

Reply:
Right I think I cover part of this when I talk about source sharing.  I'll call out how source sharing may be a strategy for avoiding public API.

Reply:
Actually this section is talking about the API used by the generated code.  I'll make sure that's clearer.